File: vademecum/code-src/zmienne_i_struktury_danych/Tablice_asocjacyjna_mapy.cpp

#include <iostream>
#include <map>

int main() {
	std::map<std::string, int> m;
	
	m["a"] = 6;
	m["cd"] = 9;
	std::cout << m["a"] << " " << m["ab"] << "\n";
	
	// wyszukanie elementu po kluczu
	std::map<std::string, int>::iterator iter = m.find("cd");
	// sprawdzenie czy istnieje
	if (iter != m.end()) {
		// wypisanie pary - klucz wartość
		std::cout << iter->first << " => " << iter->second << "\n";
		// usunięcie elementu
		m.erase(iter);
	}
	
	m["a"] = 45;
	
	// wypisanie całej mapy
	for (iter = m.begin(); iter != m.end(); ++iter)
		std::cout << iter->first << " => " << iter->second << "\n";
	// jak widać mapa jest wewnętrznie posortowana
	
	// Mapa `std::map` nie zachowuje kolejności wkładania elementów,
	// natomiast jest zawsze posortowana. C++ oferuje też inne rodzaje map, np.:
	// * nie posortowaną `std::unordered_map`,
	// * nie wymagającą unikalności klucza `std::multimap`.
}

File: vademecum/code-src/zmienne_i_struktury_danych/Tablice_asocjacyjna_mapy.py

m = { "ab" : 11, "cd" : "xx" }
x = "e"
m[x] = True;

# pobranie samych kluczy
for k in m:
	print (k, "=>", m[k])

# sprawdzenie istnienia
if "ab" in m:
	print ("jest ab")
	# usunięcie elementu
	del m['ab']

# modyfikacja wartości
m["cd"] = "oi"

# pobranie par klucz wartość
for k,v in m.items():
	print (k, "=>", v)
